cpp-restsdk code generation issue for type: string when used as  $ref:

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

The issue is the with cpp-restsdk generation  basic_string instead of utility::string_t

##### openapi-generator version

openapi-generator-cli-5.0.0

##### OpenAPI declaration file content or url

<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with

-->
```

 XYZData:
      description:
      type: object
      required:
        - events
      properties:
        events:
          type: array
          items:
            $ref: 'TSxyz.yaml#/components/schemas/Event' 
          minItems: 1
        ids:
          type: array
          items:
            type: string
          minItems: 1
        aIds:
          type: array
          items:
            type: string
          minItems: 1


    Event:
       type: string
```
  -->

##### Generation Details
java -Dlog.level=warn -jar openapi-generator-cli-5.0.0.jar generate -g cpp-restsdk -i xyz.yaml -o <file>

##### Steps to reproduce

java -Dlog.level=warn -jar openapi-generator-cli-5.0.0.jar generate -g cpp-restsdk -i xyz.yaml -o <file>

##### Related issues/PRs

This is the code generated in .h
    /// </summary>
    std::vector<std::shared_ptr<utility::string_t>>& getEvents();
    bool eventsIsSet() const;
    void unsetEvents();

    void setEvents(const std::vector<std::shared_ptr<utility::string_t>>& value);

.cpp

    if(m_EventsIsSet)
    {
        val[utility::conversions::to_string_t("Events")] = ModelBase::toJson(m_Events);
    }

./../ModelBase.h:184:23: error: no member named 'toJson' in 'std::__cxx11::basic_string<char>'
The issue is that it interprets the string as basic_string instead of utility::string_t

##### Suggest a fix

why can't we just use "basic_string" also  $ref adds a shared_ptr even for simple datatype

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestSdkClientCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestSdkClientCodegen.java
@@ -387,6 +387,8 @@ public class CppRestSdkClientCodegen extends AbstractCppCodegen {
                 || ModelUtils.isFileSchema(p) || ModelUtils.isUUIDSchema(p)
                 || languageSpecificPrimitives.contains(openAPIType)) {
             return toModelName(openAPIType);
+        } else if(typeMapping.containsKey(super.getSchemaType(p))) {
+            return openAPIType;
         }
 
         return "std::shared_ptr<" + openAPIType + ">";

